src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -46,6 +46,7 @@ private JsonPropertyInfo AddProperty(Type propertyType, PropertyInfo propertyInf
         }
 
         [PreserveDependency(".ctor()", "System.Text.Json.JsonPropertyInfoNullable`2")]
+        [PreserveDependency(".ctor()", "System.Text.Json.Serialization.JsonPropertyInfoNotNullable`4")]
```

Comment:
Fyi, this is going away in the refactoring PR here, so we should consider closing this one:
https://github.com/dotnet/runtime/pull/2259/files#diff-bcb387c5285de412415eb8d2ee62ea60L48-L50

cc @steveharter